### PR TITLE
Clear hash when `getting started` is invisible

### PR DESCRIPTION
**Pull request by HermitSun**
When I first opened this page, I clicked the 'Get Started' button and then jumped to the bottom of the page. Because I want to know more about this project, I scrolled back to the top of the page. However, when I clicked the 'Get Started' button again this time, nothing happened. That made me feel confused. Then I realized it was the problem of the anchor(#). I think others may also have this confusion, which may reduce the user experience.

So, I used `Intersection Observer` to add a listener to the homepage (downgraded to listen scroll event with throttling when the `Intersection Observer` API is not available) to remove the anchor in the URL when the part of 'Getting Started' is not visible. And then, when I click the 'Get Started' button again, I can go to the 'Getting Started' section again.

I have tested this feature on new versions of Chrome, Firefox and Edge, and it seems to work properly. But I don't have an Apple device, so I haven't tested it on Safari(and other browsers) yet.

But I'm not sure if this will conflict with the original design, and whether this will cause unnecessary performance problem to the homepage. 

And at last, sorry for my poor English.

### CODE REVIEW

**Comment by MartonDev** [comment]
Thank you for this :). We'll review it after the other pull requests have been merged

**Comment by MartonDev** [comment]
> Hi @HermitSun! First off, thank you for the PR and for noticing this. I agree that this is an issue. I think that your `Intersection Observer` is an interesting idea for solving this issue...

Yes, I agree, I haven't noticed the performance issue until you highlighted it, but it's definitely a drawback. Maybe we should consider adding [vue-smoothscroll](https://www.npmjs.com/package/vue-smoothscroll) instead? That would probably solve the issue since it is not actually "redirecting" to the hash, rather scrolling to it. It would also give a nice and fast feeling to the site. What do you think?

**Comment by HermitSun** [comment]
Thank you for your quick reply. 
I think the core of this problem is to clear the hash part(#) in URL, so that vue-router can detect the change of the anchor. That's why I use `location.hash = '_';`. `IntersectionObserver`, or scroll event listener, is just for triggering this clear behavior.

---
Oh, I think @MartonDev 's solution may be better. Detecting the change of visibility is highly cost, actually.

**Comment by MartonDev** [comment]
Yeah, another problem is (or just my concern, don't take me serious if you don't agree), that it doesn't really fit the design and structure of Vue. Thank you for your pull request anyways. If you have time and want to contribute, we'd love to see a version of this with vue-smoothscroll

**Comment by HermitSun** [comment]
> Yeah, another problem is (or just my concern, don't take me serious if you don't agree), that it doesn't really fit the design and structure of Vue. Thank you for your pull request anyways. If you have time and want to contribute, we'd love to see a version of this with vue-smoothscroll

OK, I will try. This solution is quite interesting.

**Comment by HermitSun** [comment]
It seems work well now on all my browsers, even mobile browsers.

**Comment by MartonDev** [comment]
Wow really nice! Just one thing: the get started button doesn't seem to work from other pages than the root/home

**Comment by HermitSun** [comment]
> Wow really nice! Just one thing: the get started button doesn't seem to work from other pages than the root/home

Ah, I found it just after I pushed it :-(. I think I need some time to fix it.

**Comment by HermitSun** [comment]
Maybe it can scroll to `getting started` in other pages now.
After trail and errors, I found that sometimes the page will scroll to incorrect position when click the get started button in other pages. I think it is caused by the homepage reloading when the route jumps, even if I place the scroll function in `$nextTick`. So I use `keep-alive` to cache the homepage. I’m not sure whether this is fit.

File: src/routes.js [hunk 1]
```diff
@@ -9,6 +9,9 @@ export const routes = [
   {
     path: "",
     component: Home,
+    meta: {
+      keepAlive: true
```
Comment (t8) [inline src/routes.js:13]: Not too sure about this myself. I'm interested in hearing @MartonDev's comments about this. Super creative solution @HermitSun!

Comment (t8) [reply src/routes.js:13]: @martonlederer 😉

**Comment by HermitSun** [comment]
> LGTM
> 
> We'll see how the keepalive meta affects the speed of the site

In fact, I'm not sure if the homepage being cached will affect the future changes. If the visual effect is not so important, maybe we can take `Element.scrollIntoView`  into consideration. Compatibility of this method reaches 98.5%(from [caniuse](https://caniuse.com/#search=scrollIntoView)). I used this method in my current project a few days ago.

**Comment by t8** [comment]
Interesting idea. @HermitSun, are you suggesting making more changes before this PR is merged, or would you like me to merge it now?

**Comment by HermitSun** [comment]
> Interesting idea. @HermitSun, are you suggesting making more changes before this PR is merged, or would you like me to merge it now?

I think we can merge first to see its performance. If there is any problem with the existing solution, I will try to fix it.

**Pull request merged**